File: scala/leap/src/main/scala/Leap.scala
class Year private (val year: Int) {
  def isLeap(): Boolean = {
    year match {
      case _ if year % 400 == 0 => true
      case _ if year % 100 == 0 => false
      case _ if year % 4 == 0 => true
      case _ => false
    }
  }
}

object Year {
  def apply(year: Int): Year = {
    new Year(year)
  }

}

File: csharp/leap/Leap.cs
using System;

public static class Leap
{
    public static bool IsLeapYear(int year)
    {
        bool isLeapYear = year % 4 == 0;

        if (year % 100 == 0)
        {
            isLeapYear = false;
        }

        if (year % 400 == 0)
        {
            isLeapYear = true;
        }

        return isLeapYear;
    }
}